File: presentation/04_Arrays/arrays.m
// Topic: Arrays (C-Array)
// Sum elements of an array

#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        int size = 5;
        // Using C array for fair low-level comparison, or NSMutableArray for Obj-C style
        // Let's use NSMutableArray to show object overhead vs C style
        NSMutableArray *arr = [NSMutableArray arrayWithCapacity:size];

        // Fill
        for (int i = 0; i < size; i++) {
            [arr addObject:@(i * 10)];
        }

        // Sum
        int sum = 0;
        for (NSNumber *val in arr) {
            sum += [val intValue];
        }

        NSLog(@"Sum: %d", sum);
    }
    return 0;
}

File: presentation/04_Arrays/arrays.go
// Topic: Arrays (Slices)
// Sum elements of an array

package main

import "fmt"

func main() {
	size := 5
	arr := make([]int, size)

	// Fill
	for i := 0; i < size; i++ {
		arr[i] = i * 10
	}

	// Sum
	sum := 0
	for _, val := range arr {
		sum += val
	}

	fmt.Printf("Sum: %d\n", sum)
}
